### ISSUE TITLE
Noob Question: How do I install the library, please add make 'install'

### ISSUE THREAD

**Issue by GRASBOCK** [ValveSoftware/GameNetworkingSockets#99]
I am on Linux using cmake. Was able to compile and test the example. Works!
I am used to doing make install, to install the library automatically and cleanly into the standard folders. There is no make 'install' target though. Kind of a noob at cmake, so this makes it harder.
Will try doing it manually, but I would love to see a make 'install'

**Comment by gtnardy** [comment]
If you compiled it and got the example executable, then the library binaries will be inside the build folder ;)

**Comment by GRASBOCK** [comment]
> If you compiled it and got the example executable, then the library binaries will be inside the build folder ;)

I know. I just wish there was a command that automatically installs the library to the system. Because assuming I need to reinstall, I always have to move the library manually to the /usrr/local/lib folders. I solved it using Cmake FetchContents for now. I will see if it works using standard linking next

**Comment by GRASBOCK** [comment]
I copied the staticLib.a to /usr/local/lib and the include directories into /usr/local/include. Then I linked to libGameNetworkingSockets_s.a in cmake.
While the files seem to have been found I get an error message:
```
In file included from /usr/local/include/steam/steamnetworkingtypes.h:25,
                 from /usr/local/include/steam/isteamnetworkingsockets.h:21,
                 from /usr/local/include/steam/steamnetworkingsockets.h:13,
                 from /home/.../...cc:3:
/usr/local/include/minbase/minbase_identify.h:148:3: error: #error Undefined platform
  #error Undefined platform
```

**Comment by aiusepsi** [comment]
If you want my tuppence-worth, installing the library to your system is asking for trouble. There's definitely no good reason at all to be installing the static library.

If you're already building with CMake, the simplest thing to do is to use the cmake [`add_subdirectory()`](https://cmake.org/cmake/help/latest/command/add_subdirectory.html) command to bring the GameNetworkingSockets source directory into you project. You can then link the GameNetworkingSockets::static target directly to your target.

**Comment by GRASBOCK** [comment]
Yep that worked. Easier than I thought. The reason why I wanted to install the lib on on the system is, that in case I intend to use GameNetworkingSockets across multiple projects I won't have to copy it.

Is there a way to have only one GameNetworkingSockets folder and reference it using find_package like some other cmake projects do? If not, then I will close this issue.

**Comment by ZeroByter** [comment]
Ok guys I'm just gonna hop right into this issue and bump it cause I have never used CMake before and I don't really know what I am doing.

I'm on Windows 10, I downloaded CMake 3.17.1, and using the GUI tool I tried to build the project AFTER I already added OpenSSL but now it can't find Protobuf. I tried downloading the `protoc.exe` and creating two system variables called `Protobuf_INCLUDE_DIR` and `Protobuf_LIBRARIES` which both point to the same `protoc.exe` but that didn't fix it.
To be honest that didn't really make a lot of sense but I'm getting desperate.

How to fix this Protobuf dependency issue?
